### ISSUE TITLE
Add RStudio binaries to the R PATH during installation

### ISSUE THREAD

**Issue by joelostblom** [rstudio/rstudio#11760]
RStudio bundles binaries for some programs, such as pandoc. By default these are not available when running R from the terminal since they are not added to the R PATH unless RStudio is being run. This can lead to confusing behavior, such as rmarkdown export working from RStudio but not from a terminal or even that it works *differently* in a subtle way, e.g. due to different versions of pandoc installed on the system such as in https://community.rstudio.com/t/bundled-pandoc-vs-source-pandoc/107866. This is especially confusing to students who are learning how to use R and want to export an Rmarkdown document in a reproducible way from a script.

It seems that a solution to this could be that when RStudio is installed, there is an option during the installation to add the RStudio binaries to PATH via [the relevant R configuration files](https://support.rstudio.com/hc/en-us/articles/360047157094-Managing-R-with-Rprofile-Renviron-Rprofile-site-Renviron-site-rsession-conf-and-repos-conf) that are read by R when it is run from terminal. I think this explicit option will make it clearer what is going on and save large amount of user time from troubleshooting what might be different between the RStudio environment and the R environment the terminal sees (if there is another easy way to achieve this, I would love to hear about it.

**Comment by ronblum** [comment]
@joelostblom Thank you for the suggestion! We'll review it as we continue development of RStudio.

@kevinushey Regarding the question at the end—is there currently a way to achieve this?

**Comment by kevinushey** [comment]
In this case, it's useful to look at `rmarkdown:::find_pandoc()` to see how the `rmarkdown` package resolves `pandoc`. The source is here:

https://github.com/rstudio/rmarkdown/blob/0e4f5781d54cb903f047eb7eb80425a3e6b28875/R/pandoc.R#L641-L673

The important things to note:

1. `rmarkdown` will check the paths for the `RSTUDIO_PANDOC` environment variable (normally set by RStudio), as well as a couple other locations where `pandoc` might be installed.
2. `rmarkdown` always tries to choose the newest version of `pandoc` that's installed, even if it's not first on the search list.

I think the most straightforward way to move forward here would be to ensure that `RSTUDIO_PANDOC` is set (e.g. in the user `.Renviron`?) to the path to RStudio's bundled `pandoc`?

In general, we avoid modifying the user's startup files (e.g. it would be challenging to automatically handle cases where RStudio is uninstalled, reinstalled, installed to a different location, multiple installations side-by-side, and so on).

**Comment by joelostblom** [comment]
Thanks @kevinushey ! I am leaning towards recommending our students to add the RStudio's pandoc path to `find_pandoc()` when they need to knit from terminal, even if this is an extra step each time. Setting the `RSTUDIO_PANDOC` env variable was another way I considered, and it is nice that they would only have to set it once.

**Comment by fernandoroa** [comment]
The following code fails in current version 7.1-554

```
if(system.file(package = "rmarkdown") != '') {
  if(rmarkdown::pandoc_available()) {
    if(rmarkdown::pandoc_version() > 2.11 & devel == F) {
      rmarkdown::render("www/README2.Rmd")
    }
  }
}
```
The process does not find pandoc, even having checked for it and for version.

Workaround:

```
if(system.file(package = "rmarkdown") != '') {
  if(rmarkdown::pandoc_available()) {
    if(rmarkdown::pandoc_version() > 2.11 & devel == F) {
      Sys.setenv(PATH=paste0(Sys.getenv("PATH"),":", sub("/pandoc$","",rmarkdown::pandoc_exec())))
      rmarkdown::render("www/README2.Rmd")
    }
  }
}

```

**Comment by joelostblom** [comment]
I don't think this should be closed as "stale" as there might still be things that can be done to improve the current situation.
